python/instrumentation/openinference-instrumentation-bedrock/src/openinference/instrumentation/bedrock/__init__.py

```diff
@@ -158,7 +164,99 @@ def instrumented_response(*args: Any, **kwargs: Any) -> Dict[str, Any]:
                 span.set_attributes(dict(get_attributes_from_context()))
                 return response  # type: ignore
 
-        return instrumented_response
+        @wraps(wrapped_client.converse)
+        def instrumented_response_converse(*args: Any, **kwargs: Any) -> Dict[str, Any]:
+            if context_api.get_value(_SUPPRESS_INSTRUMENTATION_KEY):
+                return wrapped_client._unwrapped_converse(*args, **kwargs)  # type: ignore
+
+            with tracer.start_as_current_span("bedrock.converse") as span:
+                span.set_attribute(
+                    SpanAttributes.OPENINFERENCE_SPAN_KIND,
+                    OpenInferenceSpanKindValues.LLM.value,
+                )
+
+                if model_id := kwargs.get("modelId"):
+                    _set_span_attribute(span, SpanAttributes.LLM_MODEL_NAME, model_id)
+
+                if inference_config := kwargs.get("inferenceConfig"):
+                    invocation_parameters = safe_json_dumps(inference_config)
+                    _set_span_attribute(
+                        span, SpanAttributes.LLM_INVOCATION_PARAMETERS, invocation_parameters
+                    )
+
+                if system_prompts := kwargs.get("system", []):
```

Comment:
small suggestion: let's process the system messages and output messages in one sequence so we don't have to worry about finicky ordering issues like needing to keep track of the message number offset